src/firmware/app/primitives/move_primitive.c

```diff
@@ -166,9 +167,10 @@ void move_start(const primitive_params_t* params, void* void_state_ptr,
 
     const FirmwareRobot_t* robot = app_firmware_world_getRobot(world);
 
-    float dx            = state->destination[0] - app_firmware_robot_getPositionX(robot);
-    float dy            = state->destination[1] - app_firmware_robot_getPositionY(robot);
-    float total_disp    = sqrtf(dx * dx + dy * dy);
+    float dx = state->destination[0] - app_firmware_robot_getPositionX(robot);
+    float dy = state->destination[1] - app_firmware_robot_getPositionY(robot);
+    // Add a small number to avoid division by zero
+    float total_disp    = sqrtf(dx * dx + dy * dy) + 1e-6;
```

Comment:
Just a thought, what if we added a `valid` flag to `state` which gets set to false if `total_disp` is 0

And then we don't add an arbitrary `1e-6`, and in the `move_tick` we do `assert(state->valid)` instead of checking
   ~~~  
   assert(!isnan(state->major_vec[0]));
    assert(!isnan(state->major_vec[1]));
    assert(!isnan(state->minor_vec[0]));
    assert(!isnan(state->minor_vec[1]));
~~~

Reply:
Yeah, I think I prefer something like this because `sqrtf(dx*dx + dy *dy)` is just as likely to be equal to `1e-6` as it is to be equal to `0`. I would perhaps suggest naming it `at_destination`. I think the `assert`'s are still OK just as general checks?

Reply:
`sqrtf(dx*dx + dy *dy)` being equal to `1e-6` is fine. It can't be negative, so adding a small epsilon is a "safe" fix in that sense.

Why add the `valid` flag to the state and then assert it later than just assert the thing in the first place?

Reply:
Yeah naming it `at_destination` makes sense, and then we avoid dividing by zero when we are "at destination"

> Why add the `valid` flag to the state and then assert it later than just assert the thing in the first place?

Thats a good point, adding an assert later on that flag doesn't make sense, the existing asserts are fine

Reply:
> sqrtf(dx*dx + dy *dy) being equal to 1e-6 is fine. It can't be negative, so adding a small epsilon is a "safe" fix in that sense.

Great point. Honestly I think given that this fix is fine as-is. Don't see a huge added benefit to adding a new field in the `state` struct for it?

Reply:
Maybe fine as a fix for now, but worth thinking about how we can write the new primitives (pending the new motion control) to be more robust and safe in this way. Eg. checking invariants, etc.